simulator-web/src/App.tsx

```diff
@@ -70,15 +90,25 @@ export const App = () => {
           <input className="text-black" value={id} onChange={(e) => void setId(e.target.value)} />
           <p className={id.length === 8 ? `text-green-500` : `text-red-500`}>{id.length} chars</p>
         </div>
+        <div className="flex flex-row gap-2">
+          <p>Fetch on focus:</p>
+          <input
+            type="checkbox"
+            className="text-black"
+            checked={fetchOnFocus}
+            onChange={() => void setFetchOnFocus(!fetchOnFocus)}
+          />
+        </div>
```

Comment:
Is there really any reason to have an option for this? I think we should just have this behaviour by default; certainly that's what I've done for the macOS app--it simply refreshes based on the app lifecycle. My goal would be to keep the UX as clean as possible. If we really want to make this configurable I'd like to push it down into a little popup with a bunch of extra stuff in the future.

Reply:
i wanted that because the decoding was not working and when it was looping forever, I didn't want the fetch to run automatically.

i guess now its kinda done, so I can remove this.